src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UISampleService.java

```diff
@@ -81,13 +88,18 @@
 	private final UICartService cartService;
 	private final MetadataEntryRepository metadataEntryRepository;
 	private final MetadataRestrictionRepository metadataRestrictionRepository;
+	private final AnalysisSubmissionRepository analysisSubmissionRepository;
+	private final UserService userService;
+	private final IridaWorkflowsService iridaWorkflowsService;
 
 	@Autowired
 	public UISampleService(SampleService sampleService, ProjectService projectService,
 			UpdateSamplePermission updateSamplePermission, SequencingObjectService sequencingObjectService,
 			GenomeAssemblyService genomeAssemblyService, MessageSource messageSource, UICartService cartService,
 			MetadataTemplateService metadataTemplateService, MetadataEntryRepository metadataEntryRepository,
-			MetadataRestrictionRepository metadataRestrictionRepository) {
+			MetadataRestrictionRepository metadataRestrictionRepository,
+			AnalysisSubmissionRepository analysisSubmissionRepository, UserService userService,
+			IridaWorkflowsService iridaWorkflowsService) {
```

Comment:
This services is getting really big, do you think we should move this stuff to maybe an UIAnalysesService?

Reply:
Yup good idea! Moved to new service in d3e0984150f4ca77d8fd65841d0c7d45ece15f41

Reply:
Awesome, thanks